src/lib/operators/table_scan/abstract_single_column_table_scan_impl.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <memory>
+#include <unordered_map>
+#include <utility>
+
+#include "abstract_table_scan_impl.hpp"
+
+#include "storage/abstract_segment_visitor.hpp"
+
+#include "types.hpp"
+
+namespace opossum {
+
+class Table;
+class ReferenceSegment;
+class AttributeVectorIterable;
+
+/**
+ * @brief The base class of table scan impls that scan a single column
+ *
+ * Resolves reference segments. The position list of reference segments
+ * is split by the referenced segments and then each is visited separately.
+ */
+class AbstractSingleColumnTableScanImpl : public AbstractTableScanImpl {
+ public:
+  AbstractSingleColumnTableScanImpl(const std::shared_ptr<const Table>& in_table, const ColumnID column_id,
+                                    const PredicateCondition predicate_condition);
+
+  std::shared_ptr<PosList> scan_chunk(const ChunkID chunk_id) const override;
+
+ protected:
+  void _scan_reference_segment(const ReferenceSegment& segment, const ChunkID chunk_id, PosList& matches) const;
+
+  // Implemented by the separate Impls. They do not need to deal with ReferenceSegments anymore, as this class
```

Comment:
All info in this comment, except the "They do not need to deal with ReferenceSegments anymore, as this class" part, is imo just adding confusion and info that the reader doesn't need to know. I'd remove most of it